data/filters/youtube-cleanup.yaml

```diff
@@ -34,6 +41,11 @@ template: |
   {{#if remove-comment-section}}
   www.youtube.com###comments #contents:remove()
   {{/if}}
+  {{#each keywords as |keyword id|}}
+  {{#if ../remove-specific-comments}}
+  www.youtube.com###comment:has-text({{keyword}}):upward(1)
```

Comment:
I'm not sure `:has-text` is the best operator here. If I understand the doc correctly, it seems like `car` would match `precarious`, but not match `Car` (with an uppercase C).

We ideally would like a case-insensitive match on full words, like the `[prop~="word"]` operator does ([see cheatsheet](https://github.com/DandelionSprout/adfilt/blob/master/Wiki/SyntaxMeaningsThatAreActuallyHumanReadable.md#element-removal-aka-cosmetic-rules-aka-hiding-rules-aka--rules)). Unfortunately, I don't think this operator cannot be applied to an element's text, but we can probably emulate it with a regexp. I'll look into it.

Reply:
You are correct that an operator cannot be applied to an element's text.
So I propose the following rule then:
`###content-text:has-text(/\b(word)\b/i):upward(ytd-comment-thread-renderer)`

Reply:
BTW, you can use the `each` shortform without naming the loop variables:

```
{{#each remove-comments-with-words}}
www.youtube.com###comment:has-text(/\b{{this}}\b/i):upward(1)
{{/each}}
```

Reply:
Don't know if the former suggested rule is better or not.

Reply:
The regexp works great, even to match a given phrase (like `great video`) :+1: You don't need the parentheses through.
About the rest of the rule, `#content-text` seems too generic to be future proof. What about prefixing the rule with `ytd-comments` to limit the search to that part of the page?

{{#each remove-comments-with-words}}
www.youtube.com##ytd-comments #content-text:has-text(/\b{{this}}\b/i):upward(ytd-comment-thread-renderer)
{{/each}}

Reply:
The CI keeps failing. I have tried different things, but it keeps failing.
You know how to solve?

Reply:
Sorry for the bumpy road @BPower0036, is there improvements to the documentation and tooling I could work on to make the contributor experience easier?

Reply:
@xvello Not that I know, the problem is that I am not good at coding at all. What I can I learned myself